airflow/utils/log/logging_mixin.py

```diff
@@ -147,18 +147,27 @@ def supports_external_link(self) -> bool:
 
 # We have to ignore typing errors here because Python I/O classes are a mess, and they do not
 # have the same type hierarchy defined as the `typing.IO` - they violate Liskov Substitution Principle
-# While it is ok to make your class derive from IOBase (and its good thing to do as they provide
+# While it is ok to make your class derive from TextIOBase (and its good thing to do as they provide
 # base implementation for IO-implementing classes, it's impossible to make them work with
 # IO generics (and apparently it has not even been intended)
 # See more: https://giters.com/python/typeshed/issues/6077
-class StreamLogWriter(IOBase, IO[str]):  # type: ignore[misc]
+class StreamLogWriter(TextIOBase, IO[str]):  # type: ignore[misc]
```

Comment:
If you would like implement some of the attribute of `TextIOBase` maybe also implements other?

```python
def writable(self):
    return True

def readable(self):
    return False

def seekable(self):
    return False

def fileno(self):
    raise OSError("Unsupported operation 'fileno'")

Reply:
I used `io.UnsupportedOperation`, which inherits from `OSError` but better describes exactly this case